Library/SharedFunctions.swift

```diff
@@ -388,3 +388,14 @@ public func isEndDateAfterToday(for reward: Reward) -> Bool {
   return (reward.endsAt == nil || (reward.endsAt ?? 0) >= AppEnvironment.current.dateType.init()
     .timeIntervalSince1970)
 }
+
+/**
+ Determines if a given `AccessTokenEnvelope`'s `User`'s email is verified.
+
+ - parameter env: The access token envelope.
+
+ - returns: A Bool representing whether the email is verified.
+ */
+public func isAccessTokenEnvelopeEmailVerified(_ env: AccessTokenEnvelope) -> Bool {
+  return env.user.isEmailVerified ?? false
```

Comment:
I think when we were moving this around we forgot to include the feature flag, it should probably be `isEmailVerified && <feature flag enabled>`? right?

Reply:
If the flag is false and `isEmailVerified` is true it's going to result in a false and showing the interstitial UI. Using similar logic with a ternary and nil-coalescing check.